src/code/snippets/src/rendering/render_pass.rs

```diff
@@ -50,18 +52,25 @@ impl MyRenderPass {
                 }
             ";
 
-        let server = renderer
-            .server
-            .as_any()
-            .downcast_ref::<GlGraphicsServer>()
-            .unwrap();
-        let shader = GlProgram::from_source(server, "MyShader", vs, fs)?;
+        let shader = renderer.server.create_program("MyShader", vs, fs)?;
+        let world_view_proj =
+            renderer
+                .server
+                .create_buffer(256, BufferKind::Uniform, BufferUsage::DynamicDraw)?;
+
+        world_view_proj.write_data_of_type(
+            &StaticUniformBuffer::<256>::new()
+                .with(&Matrix4::identity())
+                .finish(),
+        )?;
 
         Ok(Self {
             enabled: true,
-            world_view_proj: shader.uniform_location(&"worldViewProjectionMatrix".into())?,
+            world_view_proj_location: shader
```

Comment:
This location is not needed anymore, because the render pass uses uniform buffer. Instead, you need to fetch uniform buffer location. This could be done by `shader.uniform_block_index("Uniforms")`